lib/storage/src/content_manager/toc/mod.rs

```diff
@@ -71,10 +73,13 @@ pub struct TableOfContent {
     /// A lock to prevent concurrent collection creation.
     /// Effectively, this lock ensures that `create_collection` is called sequentially.
     collection_create_lock: Mutex<()>,
+    /// Dispatcher for shard transfer to access consensus.
+    shard_transfer_dispatcher: parking_lot::Mutex<Option<Dispatcher>>,
```

Comment:
`Mutex<Option<_>>` because we can only assign this after ToC is constructed.

Reply:
Btw, I just realized, if we store `Dispatcher` in the `TableOfContent`, we have an `Arc` cycle and `TableOfContent` would never be dropped. (But it might already be so, though.) Can this be an issue? 🤔

Reply:
Good catch. I think you're right.

Reply:
I'll try to see if I can integrate `Weak<T>` somewhere to prevent this.

Reply:
Would it be possible to store `Weak<TableOfContent>` and `ConsensusStateRef` inside `TableOfContent` and construct `Dispatcher` on-demand?

Reply:
> Would it be possible to store `Weak<TableOfContent>` and `ConsensusStateRef` inside `TableOfContent` and construct `Dispatcher` on-demand? (Seems even more hacky, though.)

Both can only be constructed after constructing ToC. So, both would require some `Mutex<Optional<_>>` fields in order to access them.

By using this 'dispatcher' they are grouped together.